tutorial/javascript.md

```diff
@@ -148,3 +148,201 @@ writeJSFiles = do
 ```
 
 And we're good to go. Start the server with `dist/build/tutorial/tutorial 9` and go to [http://localhost:8081/](http://localhost:8081/). Start typing in the name of one of the authors of our database or part of a book title and check out how long it takes to approximate &pi; using the method mentioned above.
+
+## Customizations
+
+Instead of calling `jquery`, you can call its variant `jqueryWith`.
+Here are the type definitions
+
+```haskell
+jquery :: JavaScriptGenerator
+jqueryWith :: CommonGeneratorOptions -> JavaScriptGenerator
+```
+
+The `CommonGeneratorOptions` will let you define different behaviors to
+change how functions are generated. Here is the definition of currently
+available options:
+
+```haskell
+data CommonGeneratorOptions = CommonGeneratorOptions
+  { 
+    -- | function transforming function names
+    functionRenamer :: String -> String
+    -- | name used when a user want to send the request body (to let you redefine it)
+  , requestBody :: String
+    -- | name of the callback parameter when the request was successful
+  , successCallback :: String
+    -- | name of the callback parameter when the request reported an error
+  , errorCallback :: String
+    -- ^ namespace on which we define the js function (empty mean local var)
+  , moduleName :: String
+  }
+```
+
+This pattern is available with all supported backends, and a default instance
```

Comment:
This sentence ends abruptly.

Probably meant to write something like: "and default values are provided".